app/src/main/java/by/academy/jc/ht/shorokhov/ht10/SpyException.java

```diff
@@ -0,0 +1,18 @@
+package by.academy.jc.ht.shorokhov.ht10;
+
+public class SpyException extends Exception {
+        public SpyException(String message) {
+                super(message);
+
+                StackTraceElement[] stack = Thread.currentThread().getStackTrace();
+                String name = stack[0].getClassName();
+                SpyStorage.setInfo(name);
```

Comment:
You had to call `SpyStorage.setInfo(name);` only in a case when exception was CATCHED not just created